### PR TITLE
Introspection beginnings

### PR DESCRIPTION
**Pull request by abays**

### CODE REVIEW

**Review by oglok — COMMENTED** [review]
Isn't this code already present in the master branch?

**Comment by abays** [comment]
Nay, it's not yet

File: pkg/server/api.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	if r.Method == http.MethodGet || r.Method == http.MethodHead {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	return
+}
+
+func (h *defaultHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (oglok) [inline pkg/server/api.go:114]: What's the difference between a defaultHandler and a healthHandler and why its http responses are different?

File: pkg/server/api.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
+package server
```
Comment (oglok) [inline pkg/server/api.go]: Although the code is quite self-explanatory, some comments would be "nice to have".

**Review by oglok — COMMENTED** [review]
Although the code is quite self-explanatory, some comments would be "nice to have" in the api.go part.

**Comment by abays** [comment]
@oglok See the pattern as established here: https://github.com/openshift/machine-config-operator/blob/master/pkg/server/api.go

**Comment by oglok** [comment]
> Nay, it's not yet

Not sure which code my previous question was referring to, but this code is the one I meant:

https://github.com/redhat-nfvpe/cluster-api-provider-baremetal/pull/19/commits/b734abb8985bd029e29836dae8382ce0b6440ae6

**Comment by abays** [comment]
Ah yes, there might be some stuff already there.  Git was acting funky when I rebased master into my branch.  The repeated code will just be a no-op therefore anyhow.

**Pull request merged**